extends TileMap
##Stae
var zielonepokojeindex = ["sala wideo","ruchomy pok贸j","pok贸j z tunelem","komora robot贸w","komora regeneracji","pusty pok贸j","sterownia"]
var 偶贸tepokojeindex = ["pok贸j z wirem","cela","ciemny pok贸j","komora chodu","komora obrotowa","komora zak贸cania","pok贸j ZRB","pok贸j luster"]
var czerwonepokojeindex = ["wanna z kwasem","zalana komora","pok贸j puapka","pok贸j iluzji","komora mierci", "pok贸j paranoi","pok贸j zegarowy","komora pi"]
var rng = RandomNumberGenerator.new()
##Zale偶ne od u偶ytkownika
var zielonepokoje = ["sala wideo","ruchomy pok贸j","sterownia","pok贸j z tunelem","pok贸j z tunelem","pok贸j z tunelem","komora robot贸w","komora robot贸w","komora regeneracji","komora regeneracji"]
var 偶贸tepokoje = ["pok贸j z wirem","pok贸j z wirem","cela","cela","ciemny pok贸j","ciemny pok贸j","komora chodu","komora chodu","komora obrotowa","komora obrotowa","komora zak贸cania","pok贸j ZRB","pok贸j ZRB","pok贸j luster"]
var czerwonepokoje = ["wanna z kwasem","wanna z kwasem","zalana komora","zalana komora","pok贸j puapka","pok贸j puapka","pok贸j iluzji","komora mierci","komora mierci", "pok贸j paranoi","pok贸j zegarowy","komora pi","komora pi"]

## Called when the node enters the scene tree for the first time.

func _ready():
	var kolumny = global.kolumny
	var szeregi = global.szeregi
	var pokoje = zielonepokoje + czerwonepokoje + 偶贸tepokoje
	##dodawanie pustych pokoi gdy normalnych jest za mao
	if pokoje.size() < kolumny*szeregi:
		for i in kolumny*szeregi - pokoje.size():
			pokoje.append("pusty pok贸j")
	randomize()
	pokoje.shuffle()
	for kolumna in kolumny:
		for szereg in szeregi:
			var tile = ""
			if kolumna == int(ceil(kolumny/2)) and szereg == int(ceil(szeregi/2)):
				tile = "pok贸j centralny"
				set_cell(0,Vector2i(kolumna,szereg),0,Vector2i(0,3))
			else:
				tile = pokoje.pop_front()
				if zielonepokojeindex.has(tile):
					set_cell(0,Vector2i(kolumna,szereg),0,Vector2i(zielonepokojeindex.find(tile,0),0))
				elif 偶贸tepokojeindex.has(tile):
					set_cell(0,Vector2i(kolumna,szereg),0,Vector2i(偶贸tepokojeindex.find(tile,0),1))
				elif czerwonepokojeindex.has(tile):
					set_cell(0,Vector2i(kolumna,szereg),0,Vector2i(czerwonepokojeindex.find(tile,0),2))
	##generacja pokoju 25 osobno
	while true:
		var kolumna = ceil(rng.randf_range(-1,kolumny-1))
		var szereg = ceil(rng.randf_range(-1,szeregi-1))
		if kolumna == 0 or kolumna == kolumny-1 or szereg == 0 or szereg == szeregi-1:
			set_cell(0,Vector2i(kolumna,szereg),0,Vector2i(1,3))
			break
